core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -82,6 +82,13 @@ class LogManager(logDirs: Seq[File],
   @volatile private var _currentDefaultConfig = initialDefaultConfig
   @volatile private var numRecoveryThreadsPerDataDir = recoveryThreadsPerDataDir
 
+  /**
+   * Lock object to protect modification to partitionsInitializing variable.
+   */
+  private val configUpdateLock = new Object
```

Comment:
Would it be more appropriate to call this `initializationLock` or something like that. It does not protect normal config updates.

Reply:
Got rid of lock as it isn't adding any value if we don't use `while` loop below. The map is concurrent hash map, so that handles concurrent updates to the key/value.